crates/pcb-sch/src/bom.rs

```diff
@@ -0,0 +1,603 @@
+
+    let mut bom_entries = Vec::new();
+
+    // Iterate through all instances and find components
+    for (instance_ref, instance) in &working_schematic.instances {
+        if instance.kind != InstanceKind::Component {
+            continue;
+        }
+
+        let designator = instance
+            .reference_designator
+            .clone()
+            .unwrap_or_else(|| format!("?{}", instance_ref.instance_path.join(".")));
+
+        // Extract attributes directly from the original map
+        let mpn = get_string_attribute(&instance.attributes, &["MPN", "Mpn", "mpn"]);
+        let manufacturer =
+            get_string_attribute(&instance.attributes, &["Manufacturer", "manufacturer"]);
+        let package = get_string_attribute(&instance.attributes, &["Package", "package"]);
+        let description =
+            get_string_attribute(&instance.attributes, &["Description", "description"]);
+        let voltage = get_string_attribute(&instance.attributes, &["Voltage"])
+            .and_then(|v| v.parse::<Value<Voltage>>().ok());
+
+        // Determine if component should be populated
+        let do_not_populate = get_string_attribute(
+            &instance.attributes,
+            &["do_not_populate", "Do_not_populate", "DNP", "dnp"],
+        )
+        .map(|s| s.to_lowercase() == "true" || s == "1")
+        .unwrap_or(false);
+
+        // Check if it's a test component
+        let is_test_component = designator.starts_with("TP")
+            || get_string_attribute(&instance.attributes, &["type", "Type"])
+                .map(|t| t.to_lowercase().contains("test"))
+                .unwrap_or(false);
+
+        let dnp = do_not_populate || is_test_component;
+
+        let value = get_string_attribute(&instance.attributes, &["Value"]);
+
+        // Extract alternates from AttributeValue::Array
+        let alternatives = instance
+            .attributes
+            .get("Alternatives")
+            .and_then(|attr| match attr {
+                AttributeValue::Array(arr) => Some(
+                    arr.iter()
+                        .filter_map(|av| match av {
+                            AttributeValue::String(s) => Some(s.clone()),
+                            _ => None,
+                        })
+                        .collect::<Vec<String>>(),
+                ),
+                _ => None,
+            })
+            .unwrap_or_default();
+
+        let well_known_module = detect_well_known_module(instance, &instance.attributes);
+
+        bom_entries.push(BomEntry {
+            designators: vec![designator],
+            mpn,
+            manufacturer,
+            alternatives,
+            package,
+            value,
+            description,
+            well_known_module,
+            dnp,
+            voltage,
+        });
+    }
+
+    // Group entries and return
+    group_bom_entries(bom_entries)
+}
+
+/// Detect well-known modules based on instance.type_ref
+fn detect_well_known_module(
```

Comment:
we used to have [`type = "resistor"`](https://github.com/diodeinc/stdlib/blob/bba910b91f6f1320df87147a42b465f876f64690/generics/Resistor.zen#L65) in the generics library but looks like it got lost at some point, this seems like the best we can do for now but we should go back to a state where you explicitly opt-in to generics resolution by specifying what type of component it is IMO (after we update the standard library). not something we need to do today though

Reply:
updated the PR to use the new "type" attribute. also, using the physical values as is, rather than parsing from string.